src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -192,38 +202,45 @@ private static boolean legsMatch(Leg previousItineraryLeg, Leg newItineraryLeg)
      * Checks whether two stops (OTP Places) match for the purposes of matching itineraries
      */
     private static boolean stopsMatch(Place stopA, Place stopB) {
-        // stop names must match
-        if (!stopA.name.equalsIgnoreCase(stopB.name)) return false;
+        // Stop names must match. It's possible in OTP to have a null place name, although it probably won't occur with
+        // transit legs. But just in case this method is expanded in scope to check more stuff about a place, if both
+        // are null, then assume a match.
+        if (
+            (stopA.name != null && !stopA.name.equalsIgnoreCase(stopB.name)) ||
+                (stopA.name == null && stopB.name != null)
+        ) {
+            return false;
+        }
 
         // stop code must match
         if (!equalsIgnoreCaseOrPreviousWasEmpty(stopA.stopCode, stopB.stopCode)) return false;
 
         // stop positions must be no further than 5 meters apart
-        if (
-            DistanceUtils.radians2Dist(
-                DistanceUtils.distHaversineRAD(stopA.lat, stopA.lon, stopB.lat, stopB.lon),
-                DistanceUtils.EARTH_MEAN_RADIUS_KM
-            ) * 1000 > 5
-        ) {
+        double stopDistanceMeters = DistanceUtils.radians2Dist(
+            DistanceUtils.distHaversineRAD(stopA.lat, stopA.lon, stopB.lat, stopB.lon),
+            DistanceUtils.EARTH_MEAN_RADIUS_KM
+        ) * 1000;
+        if (stopDistanceMeters > 5) {
             return false;
-        };
+        }
 
         // if this point is reached, the stops are assumed to match
         return true;
     }
 
     /**
-     * Returns true if previous is null. Otherwise, returns Objects.equals.
+     * Returns true if the reference value is null. Otherwise, returns Objects.equals.
      */
-    private static boolean equalsOrPreviousWasNull (Object previous, Object newer) {
-        return previous == null || Objects.equals(previous, newer);
+    private static boolean equalsOrPreviousWasNull (Object reference, Object candidate) {
```

Comment:
I think this and the below method's names might should be changed from `*Previous*` to `*Reference*`.

Reply:
Fixed in 02831be08c58fc0a47a12eb5f6927eddd76593df